src/Breadcrumbs.php

```diff
@@ -0,0 +1,312 @@
+     * simply use `$label`.
+     *
+     * Additional array elements for each link will be treated as the HTML attributes for the hyperlink tag.
+     * For example, the following link specification will generate a hyperlink with CSS class `external`:
+     *
+     * ```php
+     * [
+     *     'label' => 'demo',
+     *     'url' => 'http://example.com',
+     *     'class' => 'external',
+     * ]
+     * ```
+     *
+     * Each individual link can override global {@see encodeLabels} param like the following:
+     *
+     * ```php
+     * [
+     *     'label' => '<strong>Hello!</strong>',
+     *     'encode' => false,
+     * ]
+     * ```
+     */
+    private array $links = [];
+
+    /**
+     * @var string the template used to render each inactive item in the breadcrumbs. The token `{link}` will be
+     *             replaced with the actual HTML link for each inactive item.
+     */
+    private string $itemTemplate = "<li>{link}</li>\n";
+
+    /**
+     * @var string the template used to render each active item in the breadcrumbs. The token `{link}` will be replaced
+     *             with the actual HTML link for each active item.
+     */
+    private string $activeItemTemplate = "<li class=\"active\">{link}</li>\n";
+
+    /**
+     * Renders the widget.
+     *
+     * @return string the result of widget execution to be outputted.
+     */
+    public function run(): string
+    {
+        if (empty($this->links)) {
+            return '';
+        }
+
+        $links = [];
+
+        if ($this->homeLink === true) {
+            $links[] = $this->renderItem([
+                'label' => 'Home',
+                'url'   => '/',
+            ], $this->itemTemplate);
+        } elseif (!empty($this->homeUrlLink)) {
+            $links[] = $this->renderItem($this->homeUrlLink, $this->itemTemplate);
+        }
+
+        foreach ($this->links as $link) {
+            if (!is_array($link)) {
+                $link = ['label' => $link];
+            }
+
+            if (!empty($link)) {
+                $links[] = $this->renderItem(
+                    $link,
+                    isset($link['url']) ? $this->itemTemplate : $this->activeItemTemplate
+                );
+            }
+        }
+
+        return Html::tag(!empty($this->tag) ? $this->tag : false, implode('', $links), $this->options);
+    }
+
+    /**
+     * Renders a single breadcrumb item.
+     *
+     * @param array  $link the link to be rendered. It must contain the "label" element. The "url" element is optional.
+     * @param string $template the template to be used to rendered the link. The token "{link}" will be replaced by the
+     *                         link.
+     *
+     * @throws \InvalidArgumentException if `$link` does not have "label" element.
+     *
+     * @return string the rendering result
+     */
+    protected function renderItem(array $link, string $template): string
```

Comment:
```suggestion
    private function renderItem(array $link, string $template): string
```

Reply:
Class is extended in yii-bootstrap4, like the method.